### ISSUE TITLE
Image detection is case sensitive

### ISSUE THREAD

**Issue by dabutvin** [dabutvin/ImgBot#107]
When the image uploaded has an extension of `.PNG` instead of `.png` we respond with "No image files touched"

see https://github.com/dabutvin/ImgBot/blob/8e967e4a3cd44e520aec889114b6af2602ab71a0/ImgBot.Web/Controllers/HookController.cs#L59

we should still process the images in the repo for capitalized extensions.
This may impact the scanning for images in the worker too

This could alos use a unit test [here](https://github.com/dabutvin/ImgBot/tree/8e967e4a3cd44e520aec889114b6af2602ab71a0/ImgBot.Test)

**Comment by SuperSandro2000** [comment]
bump

**Comment by NaamloosDT** [comment]
Isn't this fixed with a simple `.ToLower()` appended to your input string?

eg
```
 .Where(file => KnownImgPatterns.ImgExtensions.Any(extension => file.ToLower().EndsWith(extension))); 
```

**Comment by dabutvin** [comment]
that should just about do it :)

We will need to update the webhook that kicks off the compressions and the actual worker that runs the compression. We should add test cases for capitalized inputs too. 

 - [Webhook](https://github.com/dabutvin/ImgBot/blob/master/WebHook/WebHookFunction.cs#L74-L76)
 - [WebHook Tests](https://github.com/dabutvin/ImgBot/blob/master/Test/WebHookTests.cs)
 - [ImageQuery](https://github.com/dabutvin/ImgBot/blob/master/CompressImagesFunction/ImageQuery.cs#L10)
 - [ImageQuery Tests](https://github.com/dabutvin/ImgBot/blob/master/Test/ImageQueryTests.cs)

Good luck!

**Comment by NaamloosDT** [comment]
👌

**Comment by guntbert** [comment]
@dabutvin about the tests: do you prefer to amend an existing test or to write a new one?

**Comment by dabutvin** [comment]
I think new tests that focus just on case sensitivity would be great.

Let me know if you need some help dropping test data in there to write the test against.

**Comment by guntbert** [comment]
My idea was to just copy `a.jpg` and `b.png` to files with uppercase extensions.
Do we need something special for mixed/upper cased filenames?

**Comment by dabutvin** [comment]
👍 uppercase extensions should be good enough 

If you want to do the webhook test too, you can copy [this file](https://github.com/dabutvin/ImgBot/blob/master/Test/data/hooks/commit-defaultbranch-images.json) to make a new test case changing `"screencap.png"` to `"screencap.PNG"`

**Comment by dabutvin** [comment]
fixed in #184
